src/backend/distributed/shared_library_init.c

```diff
@@ -2473,12 +2473,12 @@ RegisterCitusConfigVariables(void)
 		NULL, NULL, NULL);
 
 
-	DefineCustomIntVariable(
+	DefineCustomRealVariable(
 		"citus.stat_tenants_sample_rate_for_new_tenants",
```

Comment:
This new GUC should have a test too. Since it triggers random behaviour I think the only values you can realistically test are `0` and `1` but that's better than nothing at least.